### PR TITLE
Replaced caffe model with onnx for colorization sample

### PR DESCRIPTION
**Pull request by gursimarsingh**
#25006 

Improved sample for colorization with onnx model in cpp and python. Added a demo image in data folder for testing

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/dnn/colorization.cpp [hunk 1]
```diff
@@ -1,128 +1,102 @@
...
+        cerr << "The path to the ONNX model is required!" << endl;
+        return -1;
     }
 
-    Mat img = imread(imageFile);
-    if (img.empty())
-    {
-        cout << "Can't read image from file: " << imageFile << endl;
-        return 2;
+    Mat img = imread(samples::findFile(inputImagePath));
```
Comment (asmorkalov) [inline samples/dnn/colorization.cpp:49]: You can use `IMREAD_GRAYSCALE` option for `imread` and re-use lena.jpg image. No need to add the new file.

Comment (LaurentBerger) [reply samples/dnn/colorization.cpp:49]: why not use baboon.jpg, messi.jpg building.jpg?

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by LaurentBerger** [comment]
https://github.com/opencv/opencv/blob/448375d1e7b9a005b9a7e2c0c74649b54a8c36a1/samples/dnn/action_recognition.py#L48
https://github.com/opencv/opencv/blob/448375d1e7b9a005b9a7e2c0c74649b54a8c36a1/samples/dnn/classification.cpp#L105
https://github.com/opencv/opencv/blob/448375d1e7b9a005b9a7e2c0c74649b54a8c36a1/samples/dnn/edge_detection.py#L46
https://github.com/opencv/opencv/blob/448375d1e7b9a005b9a7e2c0c74649b54a8c36a1/samples/dnn/fast_neural_style.py#L16
https://github.com/opencv/opencv/blob/448375d1e7b9a005b9a7e2c0c74649b54a8c36a1/samples/dnn/object_detection.cpp#L161

**Pull request merged**